libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/IDPWebViewActivity.java

```diff
@@ -0,0 +1,171 @@
+ * without modification, are permitted provided that the following conditions
+ * are met:
+ * - Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.auth.idp;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.WindowManager;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import com.salesforce.androidsdk.R;
+import com.salesforce.androidsdk.accounts.UserAccount;
+import com.salesforce.androidsdk.util.SalesforceSDKLogger;
+import com.salesforce.androidsdk.util.UriFragmentParser;
+
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Launches a WebView, runs IDP requests within it and finishes itself when done.
+ *
+ * @author bhariharan
+ */
+public class IDPWebViewActivity extends Activity {
+
+    public static final String USER_ACCOUNT_BUNDLE_KEY = "user_account_bundle";
+    public static final String SP_CONFIG_BUNDLE_KEY = "sp_config_bundle";
+    public static final String ERROR_KEY = "error";
+    public static final String CODE_KEY = "code";
+    private static final String TAG = "IDPWebViewActivity";
+
+    private UserAccount userAccount;
+    private SPConfig spConfig;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Fetches the required extras.
+        final Intent intent = getIntent();
+        final Bundle extras = intent.getExtras();
+        if (extras != null) {
```

Comment:
Easy way to pass in `SPConfig` and `UserAccount`. Ideally, this would come from the account switcher screen based on what account was chosen (will be in the next PR).